packages/maskbook/src/plugins/RedPacket/SNSAdaptor/confirmRedPacketForm.tsx

```diff
@@ -0,0 +1,146 @@
+                    {settings?.message}
+                </Typography>
+            </Grid>
+            <Grid item xs={6}>
+                <Typography variant="body1" color="textSecondary">
+                    {t('plugin_red_packet_token')}
+                </Typography>
+            </Grid>
+            <Grid item xs={6}>
+                <Typography variant="body1" color="textPrimary" align="right">
+                    {settings?.token?.symbol}
+                    {isNative(settings?.token?.address!) ? null : (
+                        <Link
+                            className={classes.link}
+                            href={resolveTokenLinkOnExplorer(settings?.token!)}
+                            target="_blank"
+                            rel="noopener noreferrer"
+                            onClick={stop}>
+                            <LaunchIcon fontSize="small" />
+                        </Link>
+                    )}
+                </Typography>
+            </Grid>
+
+            <Grid item xs={6}>
+                <Typography variant="body1" color="textSecondary">
+                    {t('plugin_red_packet_split_mode')}
+                </Typography>
+            </Grid>
+            <Grid item xs={6}>
+                <Typography variant="body1" color="textPrimary" align="right">
+                    {settings?.isRandom ? t('plugin_red_packet_random') : t('plugin_red_packet_average')}
+                </Typography>
+            </Grid>
+
+            <Grid item xs={6}>
+                <Typography variant="body1" color="textSecondary">
+                    {t('plugin_red_packet_shares')}
+                </Typography>
+            </Grid>
+            <Grid item xs={6}>
+                <Typography variant="body1" color="textPrimary" align="right">
+                    {settings?.shares}
+                </Typography>
+            </Grid>
+
+            {settings?.isRandom ? null : (
+                <>
+                    <Grid item xs={6}>
+                        <Typography variant="body1" color="textSecondary">
+                            {t('plugin_red_packet_amount_per_share')}
+                        </Typography>
+                    </Grid>
+                    <Grid item xs={6}>
+                        <Typography variant="body1" color="textPrimary" align="right">
+                            {`${
+                                settings?.isRandom
+                                    ? formatBalance(new BigNumber(settings?.total ?? 0), settings?.token?.decimals ?? 0)
+                                    : formatBalance(
+                                          new BigNumber(settings?.total ?? 0).div(settings?.shares ?? 1),
+                                          settings?.token?.decimals ?? 18,
+                                      )
+                            } ${settings?.token?.symbol}`}
+                        </Typography>
+                    </Grid>
+                </>
+            )}
```

Comment:
There are two times of checking of `settings?.isRandom`, the last one is unnecessary, right?